app/views/legislation/questions/_votes.html.erb

```diff
@@ -0,0 +1,79 @@
+<% voted_classes = css_classes_for_vote(@legislation_question_votes, @question) %>
+<div id="<%= dom_id(@question)%>_votes" class="votes">
+  <% if user_signed_in? && current_user.level_two_verified? %>
+    <span class="in-favor">
+      <% if can?(:vote, question) %>
+        <%= link_to vote_legislation_process_question_path(question.process, question, value: "yes"),
+          method: "post", remote: true, class: "like #{voted_classes[:in_favor]}", title: t("legislation.questions.show.clear") do %>
+          <span class="icon-like"></span>
+          <span class="text"><%= t("legislation.questions.show.clear") %></span>
+        <% end %>
+      <% else %>
+        <%= link_to new_user_session_path do %>
+          <span class="icon-like"></span>
+          <span class="text"><%= t("legislation.questions.show.clear") %></span>
+        <% end %>
+      <% end %>
+    </span>
+
+    <span class="against">
+      <% if can?(:vote, question) %>
+        <%= link_to vote_legislation_process_question_path(question.process, question, value: "no"),
+          method: "post", remote: true, class: "unlike #{voted_classes[:against]}", title: t("legislation.questions.show.confusing") do %>
+          <span class="icon-unlike"></span>
+          <span class="text"><%= t("legislation.questions.show.confusing") %></span>
+        <% end %>
+      <% else %>
+        <span class="icon-unlike"></span>
+        <span class="text"><%= t("legislation.questions.show.confusing") %></span>
+      <% end %>
+    </span>
+
+  <% else %>
+
+    <div class="participation-allowed">
+      <span class="in-favor">
+        <% if can?(:vote, question) %>
+          <%= link_to vote_legislation_process_question_path(question.process, question, value: "yes"),
+              method: "post", remote: true, title: t("legislation.questions.show.clear") do %>
+              <span class="icon-like"></span>
+              <span class="text"><%= t("legislation.questions.show.clear") %></span>
+          <% end %>
+        <% else %>
+          <%= link_to new_user_session_path, class: "like" do %>
+            <span class="icon-like"></span>
+            <span class="text"><%= t("legislation.questions.show.clear") %></span>
+          <% end %>
+        <% end %>
+      </span>
+
+      <span class="against">
+        <% if can?(:vote, question) %>
+          <%= link_to vote_legislation_process_question_path(question.process, question, value: "no"),
+              method: "post", remote: true, title: t("legislation.questions.show.confusing") do %>
+              <span class="icon-unlike"></span>
+              <span class="text"><%= t("legislation.questions.show.confusing") %></span>
+          <% end %>
+        <% else %>
+          <%= link_to new_user_session_path, class: "unlike" do %>
+            <span class="icon-unlike"></span>
+            <span class="text"><%= t("legislation.questions.show.confusing") %></span>
+          <% end %>
+        <% end %>
+      </span>
```

Comment:
This whole block is the same as above, isn't it? I think in all other `_votes` partials, similar blocks are only rendered once :thinking:. Do you think it's possible to remove the duplication?

Reply:
Yeah, I realized that there is a lot of code duplication in all the `**/_votes.html.erb` views that make them very difficult to understand. I confess I was super lazy about refactoring this, but it was worth it in the end. Probably the same solution could be applied to the other `**/_votes.html.erb` views.